.github/workflows/docker_image_build.yaml

```diff
@@ -0,0 +1,22 @@
+name: Building the Docker image containing the basic tools for building ParallelZone
+on:
+  push:
+    path:
+      Docker/**
+
+jobs:
+  docker-image-build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.CONTAINER_REPO_TOKEN }}
+    - uses: actions/checkout@v2
+    - name: Build the docker image and push
```

Comment:
Generally speaking it's usually preferential to use existing actions, rather than writing our own. A quick search found [this](https://github.com/marketplace/actions/push-to-registry) for pushing to registries (and it's documentation has examples of actions for building).

Also, at some point we're hopefully going to write a paper on our CI/CD, and at that point we're going to need to note what other solutions are out there. So if you're not doing it already, you may want to take notes on what existing solutions are out there for each step and why we did/did not adopt them.

Reply:
I wrote my own action for building/pushing for two reasons:

1. I want to use my own dockerfiles, not the default "Dockerfile" under the default directory (a lot of repo image building/pushing actions only take this dockerfile).
2. Building/pushing can be easily done with two lines of commands which I have a full control for future adaptation.

Anyway, I could be wrong since I'm still very new to GitHub actions and do not have a feeling on the difference between existing and our own actions. If you prefer, I can adopt some existing building/pushing actions.

Taking notes is a very good suggestion. I will begin to that from now on. Thanks for reminding me!

Reply:
The main advantage to using existing actions is we don't have to support them, someone else does. Aside from that, they tend to integrate easier into the workflow, which makes it easier to use them in more complicated workflows.

One way or another, I think it's a good idea to get a feel for what's out there and to compare and contrast it with what we have, if for no other reason than being able to say we did our due diligence in the paper. If the existing actions simply replace one-line commands, then I don't feel like there's a need for adopting them.

Reply:
> The main advantage to using existing actions is we don't have to support them, someone else does. Aside from that, they tend to integrate easier into the workflow, which makes it easier to use them in more complicated workflows.
> 
> One way or another, I think it's a good idea to get a feel for what's out there and to compare and contrast it with what we have, if for no other reason than being able to say we did our due diligence in the paper. If the existing actions simply replace one-line commands, then I don't feel like there's a need for adopting them.

I see. Thanks for explanation. I will try to use existing actions as much as I can. No need to re-invent wheels!